redfish-core/lib/chassis.hpp

```diff
@@ -485,6 +485,35 @@ inline void handleChassisGetSubTree(
                     return; // no sensors = no failures
                 }
                 health->inventory = resp;
+
+                constexpr std::array<std::string_view, 13> interfaces = {
+                    "xyz.openbmc_project.Inventory.Item.Dimm",
+                    "xyz.openbmc_project.Inventory.Item.Cpu",
+                    "xyz.openbmc_project.Inventory.Item.PowerSupply",
+                    "xyz.openbmc_project.Inventory.Item.Fan",
+                    "xyz.openbmc_project.Inventory.Item.PCIeSlot",
+                    "xyz.openbmc_project.Inventory.Item.Vrm",
+                    "xyz.openbmc_project.Inventory.Item.Tpm",
+                    "xyz.openbmc_project.Inventory.Item.Panel",
+                    "xyz.openbmc_project.Inventory.Item.Battery",
+                    "xyz.openbmc_project.Inventory.Item.DiskBackplane",
+                    "xyz.openbmc_project.Inventory.Item.Board",
+                    "xyz.openbmc_project.Inventory.Item.Board.Motherboard",
+                    "xyz.openbmc_project.Inventory.Item.Connector"};
+                dbus::utility::getSubTreePaths(
+                    "/", 0, interfaces,
+                    [health](const boost::system::error_code& ec3,
+                             const dbus::utility::MapperGetSubTreePathsResponse&
+                                 resp2) {
+                    if (ec3)
+                    {
+                        // no inventory
+                        return;
+                    }
+
+                    health->inventory.insert(health->inventory.end(),
```

Comment:
I made the comment on upstream - https://gerrit.openbmc.org/c/openbmc/bmcweb/+/69861/comment/5831805e_1c269f7c/

----

This logic seems adding all inventory subtree paths of interfaces (even if they may not be associated with path), if there is an association of `path + "/all_sensors"?
Is it an intention?

Or, is this logic trying to find all associated subtree paths that are having all_sensors?
If so, wouldn't these chains of getAssociationEndPoints() & getSubTreePaths() be the same as getAssociatedSubtreePaths() like

```
 dbus::utility::getAssociatedSubTreePaths(
        path + "/all_sensors", "/", 0,
        interfaces,
        [health](
            const boost::system::error_code& ec,
            const dbus::utility::MapperGetSubTreePathsResponse& resp) {

        if (ec)
        {
             return; // no sensors = no failures
        }
        health->inventory = resp;
```